frontend/javascripts/oxalis/geometries/helper_geometries.ts

```diff
@@ -231,3 +314,113 @@ export class QuickSelectGeometry {
     rectangle.material.alphaMap = null;
   }
 }
+
+export class LineMeasurementGeometry {
+  color: THREE.Color;
+  line: THREE.Line<THREE.BufferGeometry, THREE.LineBasicMaterial>;
+  vertexBuffer: ResizableBuffer<Float32Array>;
+  currentOrthoView: OrthoView;
+  visible: boolean;
+  wasReset: boolean;
+
+  constructor() {
+    this.wasReset = false;
+    this.currentOrthoView = OrthoViews.PLANE_XY;
+    this.color = CONTOUR_COLOR_NORMAL;
+    this.visible = false;
+
+    const lineGeometry = new THREE.BufferGeometry();
+    const positionAttribute = new THREE.BufferAttribute(new Float32Array(3), 3);
+    positionAttribute.setUsage(THREE.DynamicDrawUsage);
+    lineGeometry.setAttribute("position", positionAttribute);
+    this.line = new THREE.Line(
+      lineGeometry,
+      new THREE.LineBasicMaterial({
+        linewidth: 2,
+      }),
+    );
+    this.line.visible = false;
+    this.line.material.color = this.color;
+    this.vertexBuffer = new ResizableBuffer(3, Float32Array);
+    this.finalizeMesh();
+  }
+
+  getMeshes() {
+    return [this.line];
+  }
+
+  reset() {
+    this.line.material.color = this.color;
+    this.vertexBuffer.clear();
+    this.finalizeMesh();
+    this.wasReset = true;
+  }
+
+  setStartPoint(pos: Vector3, initialOrthoView: OrthoView) {
+    this.wasReset = false;
+    this.visible = true;
+    this.currentOrthoView = initialOrthoView;
+    this.vertexBuffer.push(pos);
+    // Adding an additional point that will be modified by setTopPoint.
+    this.vertexBuffer.push(pos);
+    this.finalizeMesh();
+  }
+
+  setTopPoint(pos: Vector3) {
```

Comment:
I think I don't really understand what a "top point" is. maybe add a comment?

Reply:
I added a comment and renamed the method to `updateLatestPointPosition`.

By top point I meant the latest added point to the array of points (like the latest element on a stack is called top element).